### PR TITLE
feat(core): add hasher implementation info in `nx report`

### PR DESCRIPTION
**Pull request by Cammisuli**
## Current Behavior
There is no way of knowing if the workspace is using the native, git or node hasher. 

## Expected Behavior
Running `nx report` will now have information about the hasher implementation:
```
$ nx report

 >  NX   Report complete - copy this into the issue template

   Node   : 18.15.0
   OS     : darwin arm64
   yarn   : 1.22.19
   Hasher : Native
   
   nx                      : 0.0.1
   lerna                   : 6.0.0
   ....
```

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/nx/src/command-line/report.ts [hunk 1]
```diff
@@ -59,12 +63,14 @@ export async function reportHandler() {
...
   } = await getReportData();
 
   const bodyLines = [
-    `Node : ${process.versions.node}`,
-    `OS   : ${process.platform} ${process.arch}`,
-    `${pm.padEnd(5)}: ${pmVersion}`,
+    `Node   : ${process.versions.node}`,
+    `OS     : ${process.platform} ${process.arch}`,
+    `${pm.padEnd(7)}: ${pmVersion}`,
+    `hasher : ${currentHasherImplementation}`,
```
Comment (AgentEnder) [inline packages/nx/src/command-line/report.ts:73]:
```suggestion
    `Hasher : ${currentHasherImplementation}`,
```

**Review by AgentEnder — CHANGES_REQUESTED** [review]
Looks good, small formatting nitpick

**Review by AgentEnder — APPROVED** [review]

**Comment by AgentEnder** [comment]
@Cammisuli Accepting the suggestion left a bad commit message causing CI checks to fail. Can you rebase + squash locally s.t. the commitlint passes?

**Comment by Cammisuli** [comment]
dang, thats what I get when I try and be efficient 😅 I **_still_** need to do a checkout and a commit haha

**Pull request merged**